capstone-frontend/src/components/CreateClub.jsx

```diff
@@ -13,15 +13,13 @@ class CreateClub extends Component {
         data[formElements[i].name] = formElements[i].value;
       }
     }
-    fetch("/api/clubs", 
-        {method: "post", body: JSON.stringify(data)})
-        .then(function() {
-          history.push(`/clubpage/${data.id}`);
-        })
-        .catch(function(e) {
-          console.log(e);
-          alert("Looks like we're having trouble connecting to our database, hang tight!");
-        });  
+    let response = fetch("/api/clubs", {method: "post", body: JSON.stringify(data)});
+    response.catch(function(err) {
+      alert(err);
+    });
+    response.then(response => response.json()).then(response => {
+      history.push(`/clubpage/${response.id}`);
+    });
```

Comment:
Is there a way to surface this alert in the case of error without splitting it into a response variable then catching the error before you convert the json and push the new link?

If the code is invalid JSON then won't it throw an error that you could catch as in your original code and not continue to the `history.push`?

Reply:
I was a little confused as to how to set this up, as I wanted to catch errors on the fetch as well as the later conversion to json. Maybe adding another .catch() to the promise chain where I convert to json and push to history would cover the case you mentioned.

Reply:
After re-visiting the old code that I replaced, I think that did the trick better than this. I'll revert these changes.